src/qibosoq/programs/sweepers.py

```diff
@@ -32,12 +32,38 @@ def __init__(self, soc: QickSoc, qpcfg: Config, sequence: List[Pulse], qubits: L
         self.sweepers = reversed_sweepers(sweepers)
         super().__init__(soc, qpcfg, sequence, qubits)
 
+    def check_validity_sweep(self, sweeper: Sweeper):
```

Comment:
I guess the scope is clear enough, so you could even call it just
```suggestion
    def check_validity_sweep(self, sweeper: Sweeper):
```
or
```suggestion
    def validate(self, sweeper: Sweeper):
```